sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -41,14 +41,11 @@ class StringType private (val collationId: Int) extends AtomicType with Serializ
   private[sql] def supportsBinaryEquality: Boolean =
     CollationFactory.fetchCollation(collationId).supportsBinaryEquality
 
-  private[sql] def supportsLowercaseEquality: Boolean =
-    CollationFactory.fetchCollation(collationId).supportsLowercaseEquality
```

Comment:
why are we removing this?

Reply:
Not used anywhere.

Reply:
maybe not at the moment, but I recommend that we keep it for private SQL datatype API completeness

let's not remove it for now, especially given that this change doesn't have anything to do with this PR